dbms/src/Common/UniThreadPool.h

```diff
@@ -93,6 +98,12 @@ class ThreadPoolImpl
     void setQueueSize(size_t value);
     size_t getMaxThreads() const;
 
+    std::shared_ptr<ThreadPoolWaitGroup<Thread>> waitGroup()
+    {
+        std::shared_ptr<ThreadPoolWaitGroup<Thread>> wait_group = std::make_shared<ThreadPoolWaitGroup<Thread>>(*this);
```

Comment:
ThreadPoolWaitGroup is **not** shareable because it is not thread safe. You may use unique_ptr instead.